test/functional/legacy/019_smarttab_expandtab_spec.lua

```diff
@@ -0,0 +1,56 @@
+-- Tests for "r<Tab>" with 'smarttab' and 'expandtab' set/not set.
+-- Also test that dv_ works correctly
+
+local helpers = require('test.functional.helpers')
+local feed, insert = helpers.feed, helpers.insert
+local clear, execute, expect = helpers.clear, helpers.execute, helpers.expect
+
+describe('performing "r<Tab>" with \'smarttab\' and \'expandtab\' set/not set, and "dv_"', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([[
+      start text
+      		some test text
+      test text
+      		other test text
+          a cde
+          f ghi
+      test text
+        Second line beginning with whitespace]])
+
+    execute('set smarttab expandtab ts=8 sw=4')
+    -- Make sure that backspace works, no matter what termcap is used.
+    execute('set t_kD=x7f t_kb=x08')
```

Comment:
I am not sure if `set t_kD` makes sense for us anymore. cc @tarruda 

And, original test has `set t_kD=^Vx7f t_kb=^Vx08 `. This means CTRL-V should be fed before x7f, etc...